ManyToOneRel fields (reverse FK) are broken in 1.3.3

I'm trying to make an object with M2O field value set to a list of unsaved objects with FK set to `None` (see example below for details). This worked fine with 1.3.2 but is broken in 1.3.3 and 1.4.0

## Expected behavior

`baker.make` with `M2O` fields  correctly saves parent object and sets FK to child objects before saving them

## Actual behavior

Fails with `NOT NULL constraint failed` exception

## Reproduction Steps

I have the following models structure

```python
class Foo(models.Model):
    pass


class Bar(models.Model):
    foos = models.ManyToManyField(Foo)


class Baz(models.Model):
    bar = models.ForeignKey(Bar, related_name="bazs", on_delete=models.CASCADE)
```

With 1.3.2 the following test works just fine ([ac48f878](https://github.com/model-bakers/model_bakery/commit/ac48f878))

```python
@pytest.mark.django_db
class TestsFoo:
    def test_foo(self):
        baz1, baz2 = baker.prepare(models.Baz, bar=None, _quantity=2)
        baker.make(
            models.Bar,
            foos=[baker.make(models.Foo)],
            bazs=[baz1, baz2],
        )
```

With 1.3.3 ([5d1be500](https://github.com/model-bakers/model_bakery/commit/5d1be500)) it fails with the following error:

```text
tests/test_foo.py:8 (TestsFoo.test_foo)
../venv/lib/python3.9/site-packages/django/db/backends/utils.py:85: in _execute
    return self.cursor.execute(sql, params)
../venv/lib/python3.9/site-packages/django/db/backends/sqlite3/base.py:416: in execute
    return Database.Cursor.execute(self, query, params)
E   sqlite3.IntegrityError: NOT NULL constraint failed: generic_baz.bar_id

The above exception was the direct cause of the following exception:
test_foo.py:11: in test_foo
    baker.make(
../model_bakery/baker.py:98: in make
    return baker.make(
../model_bakery/baker.py:324: in make
    return self._make(**params)
../model_bakery/baker.py:384: in _make
    instance = self.instance(
../model_bakery/baker.py:419: in instance
    self._handle_one_to_many(instance, one_to_many_keys)
../model_bakery/baker.py:516: in _handle_one_to_many
    value.save()
../venv/lib/python3.9/site-packages/django/db/models/base.py:743: in save
    self.save_base(using=using, force_insert=force_insert,
../venv/lib/python3.9/site-packages/django/db/models/base.py:780: in save_base
    updated = self._save_table(
../venv/lib/python3.9/site-packages/django/db/models/base.py:885: in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
../venv/lib/python3.9/site-packages/django/db/models/base.py:923: in _do_insert
    return manager._insert(
../venv/lib/python3.9/site-packages/django/db/models/manager.py:85: in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
../venv/lib/python3.9/site-packages/django/db/models/query.py:1301: in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
../venv/lib/python3.9/site-packages/django/db/models/sql/compiler.py:1441: in execute_sql
    cursor.execute(sql, params)
../venv/lib/python3.9/site-packages/django/db/backends/utils.py:67: in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
../venv/lib/python3.9/site-packages/django/db/backends/utils.py:76: in _execute_with_wrappers
    return executor(sql, params, many, context)
../venv/lib/python3.9/site-packages/django/db/backends/utils.py:85: in _execute
    return self.cursor.execute(sql, params)
../venv/lib/python3.9/site-packages/django/db/utils.py:90: in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
../venv/lib/python3.9/site-packages/django/db/backends/utils.py:85: in _execute
    return self.cursor.execute(sql, params)
../venv/lib/python3.9/site-packages/django/db/backends/sqlite3/base.py:416: in execute
    return Database.Cursor.execute(self, query, params)
E   django.db.utils.IntegrityError: NOT NULL constraint failed: generic_baz.bar_id
```

Fails because [here](https://github.com/model-bakers/model_bakery/commit/5d1be500#diff-e5857deb915e241f429a0c118e89e06a3388d3ce1466e3aa4b960b7055172b6dR514-R516) we're trying to save M2O objects with `FK` set to `None`. 

### Versions

Python: 3.9.10
Django: 4.0.2
Model Bakery: 1.3.3

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Extend type hints in `model_bakery.recipe` module, make `Recipe` class generic [PR #292](https://github.com/model-bakers/model_bakery/pull/292)
 - Explicitly add _fill_optional parameters to baker.make and baker.prepare to aid IDE autocomplete function. [PR #264](https://github.com/model-bakers/model_bakery/pull/264)
 - Fixed errors with reverse M2M relationships [PR #299](https://github.com/model-bakers/model_bakery/pull/299)
+- Fixed errors with reverse M2O relationships [PR #300](https://github.com/model-bakers/model_bakery/pull/300)
 
 ### Removed
 

--- a/model_bakery/baker.py
+++ b/model_bakery/baker.py
@@ -609,11 +609,13 @@ class Baker(Generic[M]):
                 # Django will handle any operation to persist nested non-persisted FK because
                 # save don't do so and, thus, raise constraint errors. That's why save() only gets
                 # called if the object doesn't have a pk and also doesn't hold fk pointers
-                fks = any([
-                    fk
-                    for fk in value._meta.fields
-                    if isinstance(fk, ForeignKey) or isinstance(fk, OneToOneField)
-                ])
+                fks = any(
+                    [
+                        fk
+                        for fk in value._meta.fields
+                        if isinstance(fk, ForeignKey) or isinstance(fk, OneToOneField)
+                    ]
+                )
                 if not value.pk and not fks:
                     value.save()
 

--- a/model_bakery/baker.py
+++ b/model_bakery/baker.py
@@ -607,8 +607,9 @@ class Baker(Generic[M]):
 
             for value in values:
                 # Django will handle any operation to persist nested non-persisted FK because
-                # save doesn't do so and, thus, raises constraint errors. That's why save() only gets
-                # called if the object doesn't have a pk and also doesn't hold fk pointers
+                # save doesn't do so and, thus, raises constraint errors. That's why save()
+                # only gets called if the object doesn't have a pk and also doesn't hold fk
+                # pointers.
                 fks = any(
                     [
                         fk

--- a/model_bakery/baker.py
+++ b/model_bakery/baker.py
@@ -607,7 +607,7 @@ class Baker(Generic[M]):
 
             for value in values:
                 # Django will handle any operation to persist nested non-persisted FK because
-                # save don't do so and, thus, raise constraint errors. That's why save() only gets
+                # save doesn't do so and, thus, raises constraint errors. That's why save() only gets
                 # called if the object doesn't have a pk and also doesn't hold fk pointers
                 fks = any(
                     [

--- a/model_bakery/baker.py
+++ b/model_bakery/baker.py
@@ -606,7 +606,15 @@ class Baker(Generic[M]):
             manager = getattr(instance, key)
 
             for value in values:
-                if not value.pk:
+                # Django will handle any operation to persist nested non-persisted FK because
+                # save don't do so and, thus, raise constraint errors. That's why save() only gets
+                # called if the object doesn't have a pk and also doesn't hold fk pointers
+                fks = any([
+                    fk
+                    for fk in value._meta.fields
+                    if isinstance(fk, ForeignKey) or isinstance(fk, OneToOneField)
+                ])
+                if not value.pk and not fks:
                     value.save()
 
             try:

--- a/tests/generic/models.py
+++ b/tests/generic/models.py
@@ -457,7 +457,7 @@ class NonStandardManager(models.Model):
     manager = models.Manager()
 
 
-### The followin models were added after issue 291
+# The followin models were added after issue 291
 # Since they doesn't hold much meaning, they are only numbered ones
 class Issue291Model1(models.Model):
     pass
@@ -468,5 +468,7 @@ class Issue291Model2(models.Model):
 
 
 class Issue291Model3(models.Model):
-    fk_model_2 = models.ForeignKey(Issue291Model2, related_name="bazs", on_delete=models.CASCADE)
+    fk_model_2 = models.ForeignKey(
+        Issue291Model2, related_name="bazs", on_delete=models.CASCADE
+    )
     name = models.CharField(max_length=32)

--- a/tests/generic/models.py
+++ b/tests/generic/models.py
@@ -458,7 +458,7 @@ class NonStandardManager(models.Model):
 
 
 # The followin models were added after issue 291
-# Since they doesn't hold much meaning, they are only numbered ones
+# Since they don't hold much meaning, they are only numbered ones
 class Issue291Model1(models.Model):
     pass
 

--- a/tests/generic/models.py
+++ b/tests/generic/models.py
@@ -455,3 +455,18 @@ class NonStandardManager(models.Model):
     name = models.CharField(max_length=30)
 
     manager = models.Manager()
+
+
+### The followin models were added after issue 291
+# Since they doesn't hold much meaning, they are only numbered ones
+class Issue291Model1(models.Model):
+    pass
+
+
+class Issue291Model2(models.Model):
+    m2m_model_1 = models.ManyToManyField(Issue291Model1)
+
+
+class Issue291Model3(models.Model):
+    fk_model_2 = models.ForeignKey(Issue291Model2, related_name="bazs", on_delete=models.CASCADE)
+    name = models.CharField(max_length=32)

--- a/tests/test_baker.py
+++ b/tests/test_baker.py
@@ -571,7 +571,9 @@ class TestBakerCreatesAssociatedModels(TestCase):
 
     def test_ensure_reverse_fk_for_many_to_one_is_working(self):
         """This is a regression test to make sure issue 291 is fixed"""
-        fk1, fk2 = baker.prepare(models.Issue291Model3, fk_model_2=None, name="custom name", _quantity=2)
+        fk1, fk2 = baker.prepare(
+            models.Issue291Model3, fk_model_2=None, name="custom name", _quantity=2
+        )
         obj = baker.make(
             models.Issue291Model2,
             m2m_model_1=[baker.make(models.Issue291Model1)],

--- a/tests/test_baker.py
+++ b/tests/test_baker.py
@@ -569,6 +569,20 @@ class TestBakerCreatesAssociatedModels(TestCase):
         assert not person.pk
         assert 0 == models.RelatedNamesModel.objects.count()
 
+    def test_ensure_reverse_fk_for_many_to_one_is_working(self):
+        """This is a regression test to make sure issue 291 is fixed"""
+        fk1, fk2 = baker.prepare(models.Issue291Model3, fk_model_2=None, name="custom name", _quantity=2)
+        obj = baker.make(
+            models.Issue291Model2,
+            m2m_model_1=[baker.make(models.Issue291Model1)],
+            bazs=[fk1, fk2],
+        )
+
+        assert obj.bazs.count() == 2
+        related_1, related_2 = obj.bazs.all()
+        assert related_1.name == "custom name"
+        assert related_2.name == "custom name"
+
 
 @pytest.mark.django_db
 class TestHandlingUnsupportedModels:

